### DISCUSSION TITLE
Question on TFT chip and display driver discrepencies

### DISCUSSION THREAD

**Discussion by shan-mcarthur** [Bodmer/TFT_eSPI#2553] [Q&A - General]
I have bought a TFT that has an ILI9488 display on it.  When I use the Read_User_setup example, It is outputting the display driver being 9341 and not 9844.  my header defines ILI9488_DRIVER and not ILI9341_DRIVER.  Is this a problem, or does my board not have the right chip on it?  It is a 3.5" display by my caliper, and the resolution is reporting 320x480, so it feels like the ILI9488.  The output below is from the results of the tft.getSetup() call.  What is causing the library to think it is a 9341?

[code]
TFT_eSPI ver = 2.5.0
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
**Display driver = 9341**
Display width  = 320
Display height = 480

MOSI    = GPIO 23
SCK     = GPIO 18
TFT_CS   = GPIO 15
TFT_DC   = GPIO 0
TFT_RST  = GPIO 2

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]

**Comment by Bodmer** [comment]
I suspect line 30 in User_Setup.h has not been commented out, so the default setup is taking precedence.

**Reply by shan-mcarthur** [reply]
I have #define USER_SETUP_LOADED at the top of my sketch, before the include for TFT_eSPI.  Line 21 in the User_Setup_Select.h should skip the USer_Setup.h include because it is already defined.

**Reply by Bodmer** [reply]
That will not work. Any #define in the sketch  or in headers called up by the sketch will not be visible to the library. So use the recommended approach.
